app/services/orchestration_template_dialog_service.rb

```diff
@@ -213,7 +213,7 @@ def add_parameter_field(parameter, group, position)
   end
 
   def create_parameter_dropdown_list(parameter, group, position, dropdown)
-    dropdown_list = dropdown.allowed_values.collect { |v| [v, v] }
+    dropdown_list = dropdown.allowed_values.collect { |v| v.kind_of?(Hash) ? [v.keys.first, v[v.keys.first]] : [v, v] }
```

Comment:
As you can see in the spec, the alowed values is always an array. This array can contain either simple hash `{ key => value }` or simple values (string). This allows one to use such an array:

`["val1", { "key2" => "val2" }, "val3" ]`

I considered the option you propose, but chose "mine". One reason being that I would still like to support setting the order of the values.